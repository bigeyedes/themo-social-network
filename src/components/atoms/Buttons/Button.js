import styled, { css } from 'styled-components'

const Button = styled.button`
	background: #3975F5;
	color: white;
	border: none;
	border-radius: 5px;
	padding: 10px 20px;
	margin: 10px;
	transition: .3s ease;
	cursor: pointer;
	&:hover {
		opacity: .7
	}

	${({ secondary }) =>
    secondary &&
    css`
      background-color: black;
    `}
`

export default Button