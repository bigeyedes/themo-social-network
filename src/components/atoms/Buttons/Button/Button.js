import styled, { css } from 'styled-components'

const Button = styled.button`
	background: #3975F5;
	color: white;
	border: none;
	border-radius: 5px;
	padding: 10px 20px;
	margin: 10px;
	transition: .3s ease;
	cursor: pointer;
	&:hover {
		opacity: 0.7
	}

	${({ secondary }) =>
    secondary &&
    css`
      background: black;
    `}
`

export default Button