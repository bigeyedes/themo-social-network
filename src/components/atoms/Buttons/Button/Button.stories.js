import React from 'react'
import { storiesOf } from '@storybook/react';
import Button from './Button';

storiesOf('Atoms/Buttons', module)
	.add('Primary Button', () => <Button>Primary</Button>)
	.add('Secondary Button', () => <Button secondary>Secondary</Button>)