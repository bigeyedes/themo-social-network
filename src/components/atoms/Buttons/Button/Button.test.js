import React from 'react'
import Button from './Button'
import renderer from 'react-test-renderer';
import 'jest-styled-components'

describe("Render button component", () => {
	test("Matches the snapshot", () => {
		const tree = renderer.create(<Button />).toJSON()
		expect(tree).toMatchSnapshot()
	});

	test("If is 0.7 opacity when hover", () => {
		const tree = renderer.create(<Button />).toJSON();
		expect(tree).toHaveStyleRule('opacity', '0.7', {
			modifier: ':hover',
		  })
	});

	test("If is black background when secondary props is passed", () => {
		const tree = renderer.create(<Button secondary></Button>).toJSON();
		expect(tree).toHaveStyleRule('background', 'black')
	});

});