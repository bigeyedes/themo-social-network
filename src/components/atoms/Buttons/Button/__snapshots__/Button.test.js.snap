// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render button component Matches the snapshot 1`] = `
.c0 {
  background: #3975F5;
  color: white;
  border: none;
  border-radius: 5px;
  padding: 10px 20px;
  margin: 10px;
  -webkit-transition: .3s ease;
  transition: .3s ease;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

<button
  className="c0"
/>
`;
